x/ibc/light-clients/solomachine/types/client_state.go

```diff
@@ -0,0 +1,409 @@
+package types
+
+import (
+	ics23 "github.com/confio/ics23/go"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	clientexported "github.com/cosmos/cosmos-sdk/x/ibc/02-client/exported"
+	clienttypes "github.com/cosmos/cosmos-sdk/x/ibc/02-client/types"
+	connectionexported "github.com/cosmos/cosmos-sdk/x/ibc/03-connection/exported"
+	channelexported "github.com/cosmos/cosmos-sdk/x/ibc/04-channel/exported"
+	commitmentexported "github.com/cosmos/cosmos-sdk/x/ibc/23-commitment/exported"
+	commitmenttypes "github.com/cosmos/cosmos-sdk/x/ibc/23-commitment/types"
+	host "github.com/cosmos/cosmos-sdk/x/ibc/24-host"
+)
+
+var _ clientexported.ClientState = (*ClientState)(nil)
+
+// NewClientState creates a new ClientState instance.
+func NewClientState(chainID string, consensusState *ConsensusState) *ClientState {
+	return &ClientState{
+		ChainId:        chainID,
+		FrozenHeight:   0,
+		ConsensusState: consensusState,
+	}
+}
+
+// GetChainID returns an empty string.
```

Comment:
Agreed (also see above). Sorry for my confusing comment before. Why do we have this interface function in the first place?

Reply:
It seems there is partial consensus on removing ChainID. cc/ @fedekunze

Reply:
good question. I don't think it is being used outside of tendermint (but it can remain a tm client state specific func). I think we can just remove it in another pr

Reply:
opened #7143